bin/lint-markdown-api-history.ts

```diff
@@ -0,0 +1,616 @@
+  breakingChangesFile: string;
+  checkStrings: boolean;
+  ignoreGlobs: string[];
+  schema: string;
+}
+
+interface HTMLWithPreviousNode extends HTML {
+  previousNode?: Node;
+}
+
+// If you change this, you might want to update the one in the website transformer
+export interface PrReleaseVersions {
+  release: string | null;
+  backports: Array<string>;
+}
+
+// If you change this, you might want to update the one in the website transformer
+export type PrReleaseVersionsContainer = { [key: number]: PrReleaseVersions };
+
+// If you change this, you might want to update the one in the website transformer
+interface PrReleaseArtifact {
+  data: PrReleaseVersionsContainer;
+  endCursor: string;
+}
+
+function getCIPrNumber(): number | null {
+  if (process.env.GITHUB_REF_NAME?.endsWith('/merge')) {
+    // https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables
+    return Number(process.env.GITHUB_REF_NAME.split('/')[0]);
+  } else if (process.env.CIRCLE_PULL_REQUEST?.includes('/pull/')) {
+    // https://circleci.com/docs/variables/#built-in-environment-variables
+    return Number(process.env.CIRCLE_PULL_REQUEST.split('/').at(-1));
+  } else {
+    return null;
+  }
+}
+
+let _allPrReleaseVersions: PrReleaseVersionsContainer;
+
+// If you change this, you might want to update the one in the website transformer
+// TODO: Change this when GH_TOKEN isn't needed to fetch PR release versions anymore
+async function getAllPrReleaseVersions(): Promise<PrReleaseVersionsContainer> {
+  try {
+    if (_allPrReleaseVersions) {
+      return _allPrReleaseVersions;
+    }
+
+    if (process.env.NODE_ENV === 'test') {
+      const versions: PrReleaseVersionsContainer = {
+        22533: {
+          release: '',
+          backports: [] as string[],
+        },
+        26789: {
+          release: '',
+          backports: [] as string[],
+        },
+        37094: {
+          release: '',
+          backports: [] as string[],
+        },
+      };
+
+      _allPrReleaseVersions = versions;
+
+      const ciPrNumber = getCIPrNumber();
+      if (ciPrNumber) {
+        console.log(
+          'Detected PR number:\n\n' +
+            `'${ciPrNumber}'\n\n` +
+            'using CI environment variables. Adding to list of PRs...\n',
+        );
+        _allPrReleaseVersions[ciPrNumber] = {
+          release: '',
+          backports: [],
+        };
+      }
+
+      return _allPrReleaseVersions;
+    }
+
+    if (!process.env.GH_TOKEN) {
+      throw new Error(`GH_TOKEN is required for fetching PR release versions.`);
+    }
+
+    const fetchOptions = {
+      method: 'GET',
+      headers: {
+        Accept: 'application/vnd.github+json',
+        'X-GitHub-Api-Version': '2022-11-28',
+        Authorization: `Bearer ${process.env.GH_TOKEN}`,
+      },
+    };
+
+    const artifactsListResponse = await fetch(
+      'https://api.github.com/repos/electron/website/actions/artifacts',
```

Comment:
Nit: In the areas where we have this hardcoded, let's just make a constant at the top of the file, so we can easily find this and can reuse it in other areas if needed. We do something similar here: https://github.com/electron/electron/blob/main/script/release/ci-release-build.js#L13